rai/node/node.cpp

```diff
@@ -2337,12 +2369,11 @@ class confirmed_visitor : public rai::block_visitor
 void rai::node::process_confirmed (std::shared_ptr<rai::block> block_a)
 {
 	auto hash (block_a->hash ());
-	bool exists (ledger.block_exists (hash));
-	// Attempt to process confirmed block if it's not in ledger yet
-	if (!exists)
+	bool exists;
 	{
+		// We always re-process the block to guarantee it's marked as confirmed
```

Comment:
Wouldn't this result in terrible performance? Single write transaction for each confirmed block

Reply:
Good point, though it could cause a lot of problems if block_confirmed sometimes returned false for blocks from the callback.

Reply:
Maybe I could move the block observer call into process_return_one after processing a confirmed block. That'd actually make a lot of sense.